web/pgadmin/tools/erd/static/js/erd_tool/components/ERDTool.jsx

```diff
@@ -369,10 +357,9 @@ export default class ERDTool extends React.Component {
 
 
     if(this.props.params.sql_id){
-      let sqlValue = localStorage.getItem(this.props.params.sql_id);
-      localStorage.removeItem(this.props.params.sql_id);
+      let sqlValue = retrieveDataFromLocalStorgae(this.props.params.sql_id);
       if (sqlValue) {
-        this.diagram.deserialize((JSON.parse(sqlValue)));
```

Comment:
why is JSON parse removed?

Reply:
While returning data from retrieveDataFromLocalStorgae is already parsed.